docs/src/operations/authorization.md

```diff
@@ -32,32 +44,150 @@ public class SecureClient {
 }
 ```
 
-It should be noted that the CredentialsProvider interface is unstable since the authorization features of Zeebe are still under development. 
 
-> **Note:** Zeebe's Java client will not prevent you from adding credentials to gRPC calls while using an insecure connection but you should be aware of the risks of doing so. The client will log a warning message whenever such a call is made, to prevent this from happening accidentaly. 
-
-## Go
-
-Adding credentials to gRPC headers in the client works in a similar way to the Java client. The CredentialsProvider interface contains a single method, `ApplyCredentials(headers map[string]string)` that takes a map of headers to which it should add access credentials:
+### Go
 
 ```go
 type MyCredentialsProvider struct {
-	token string
 }
 
-func (provider MyCredentialsProvider) ApplyCredentials(headers map[string]string) {
-	headers["Authorization"] = provider.token
+// ApplyCredentials adds a token to the Authorization header of a gRPC call.
+func (p *MyCredentialsProvider) ApplyCredentials(headers map[string]string) {
+	headers["Authorization"] = "someToken"
 }
 
-func main() {
-	provider := &MyCredentialsProvider{token: "someKindOfToken"}
+// ShouldRetryRequest returns true if the call failed with a deadline exceed error.
+func (p *MyCredentialsProvider) ShouldRetryRequest(err error) bool {
+  return status.Code(err) == codes.DeadlineExceeded
+}
 
+func main() {
 	client, err := NewZBClient(&ZBClientConfig{
-		CredentialsProvider:  provider,
+		CredentialsProvider:  &MyCredentialsProvider{},
 	})
+  if err != nil {
+    panic(err)
+  }
   
-  // continue... 
+  response, err := client.NewTopologyCommand().Send()
+  if err != nil {
+    panic(err)
+  }
+    
+  fmt.Println(response.String())
+}
+```
+## OAuthCredentialsProvider
+
+The OAuthCredentialsProvider requires the specification of a client ID and a client secret. These are then used to request an access token from an OAuth 2.0 authorization server through a [client credentials flow](https://tools.ietf.org/html/rfc6749#section-4.4). By default, the authorization server is the one used by Camunda Cloud but any other can be used. Using the access token returned by the authorization server, the OAuthCredentialsProvider will add it to the gRPC headers of each request as a bearer token. If a requests fails with an UNAUTHENTICATED gRPC code and if the stored access token can be refreshed, the request will be retried. If the failure is caused by any other code or if the refreshed access token is the same as the stale access token, then the request will not be retried.
```

Comment:
I feel like the following could be worded in an simpler way:

> If a requests fails with an UNAUTHENTICATED gRPC code and if the stored access token can be refreshed, the request will be retried. If the failure is caused by any other code or if the refreshed access token is the same as the stale access token, then the request will not be retried.

Maybe something like:

> Requests which fail with an `UNAUTHENTICATED` gRPC code are seamlessly retried if and only if a new access token can be obtained.

We could even split it from the paragraph and add it as a block note. What do you think?

Reply:
I agree, that's better. I don't see a reason to use a blockquote but I suppose it's just a personal preference/style thing

Reply:
It's fine in the same paragraph